index.js

```diff
@@ -31,20 +34,33 @@ class IndexManipulation {
       this.sourceApplicationID,
       this.sourceApiKey
     );
+    Object.keys(this.headers).forEach(key => {
+      sourceClient.setExtraHeader(key, this.headers[key]);
+    });
     this.sourceIndex = sourceClient.initIndex(this.sourceIndexName);
 
     const destinationClient = algoliasearch(
       this.destinationApplicationID,
       this.destinationApiKey
     );
+    Object.keys(this.headers).forEach(key => {
+      destinationClient.setExtraHeader(key, this.headers[key]);
+    });
     this.destinationIndex = destinationClient.initIndex(
       this.destinationIndexName
     );
   }
 
   _writeSourceChunks() {
     return new Promise((resolve, reject) => {
-      const browser = this.sourceIndex.browseAll();
+      const browser = this.sourceIndex.browseAll(
+        Object.assign(
+          {
+            attributesToRetrieve: '*'
```

Comment:
Just to make sure: why asking to retrieve all the attributes by default?

Reply:
When moving an index you normally want all fields because you want to work with them (Optimisation of attributes etc). You can overwrite this manually if you choose